packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/TypeConverterProvider.kt

```diff
@@ -68,9 +69,15 @@ object TypeConverterProviderImpl : TypeConverterProvider {
       return EnumTypeConverter(kClass as KClass<Enum<*>>, type.isMarkedNullable)
     }
 
-    // TODO(@lukmccall): cache record type converters
+    val maybeConverter = cachedRecordConverters[kClass]
+    if (maybeConverter != null) {
+      return maybeConverter
+    }
+
     if (kClass.isSubclassOf(Record::class)) {
-      return RecordTypeConverter<Record>(this, type)
+      val converter = RecordTypeConverter<Record>(this, type)
```

Comment:
Is the creation of the record type converter so expensive to be worth caching?

Reply:
Nope, but we can easily reuse converters. For instance, if you have two methods in the module which uses the same `Record`, we don't have to create a separate object for them. In that case, we save a little bit of memory. Currently, records converters hold all the needed information to create a Kotlin object. In that way, we have an extremely fast runtime with not that huge memory overhead. IMO, it's worth caching.

Reply:
TBH, I would like to see some benchmarks 😅 
I get that it's super awesome when we do optimise the code, but I don't see the reason for doing so if the more naive way is performant enough 🤔